compiler/rustc_lint/src/noop_method_call.rs

```diff
@@ -135,7 +137,9 @@ impl<'tcx> LateLintPass<'tcx> for NoopMethodCall {
                     span,
                     SuspiciousDoubleRefDerefDiag { ty: expr_ty },
                 ),
-                _ => return,
+                // If `type_of(x) == T` and `x.borrow()` is used to get `&T`,
+                // then that should be allowed
+                sym::noop_method_borrow | _ => return,
```

Comment:
I don't think this change is necessarily better than the code before.

Reply:
That really shouldn't have been in there. It's not really better or worse.